tests/integration/nfs_test.go

```diff
@@ -0,0 +1,147 @@
+	suite.TearDown()
+}
+
+func (suite *NfsSuite) SetUp() {
+	suite.namespace = "nfs-ns"
+	suite.systemNamespace = installer.SystemNamespace(suite.namespace)
+	suite.instanceCount = 1
+
+	k8sHelper, err := utils.CreateK8sHelper(suite.T)
+	require.NoError(suite.T(), err)
+	suite.k8sHelper = k8sHelper
+
+	k8sversion := suite.k8sHelper.GetK8sServerVersion()
+	logger.Infof("Installing nfs server on k8s %s", k8sversion)
+
+	suite.installHelper = installer.NewK8sRookhelper(suite.k8sHelper.Clientset, suite.T)
+
+	suite.rwClient = clients.CreateReadWriteOperation(suite.k8sHelper)
+
+	err = suite.installHelper.InstallNFSServer(suite.systemNamespace, suite.namespace, suite.instanceCount)
+	if err != nil {
+		logger.Errorf("nfs server installation failed: %+v", err)
+		suite.T().Fail()
+		suite.TearDown()
+		suite.T().FailNow()
+	}
+}
+
+func (suite *NfsSuite) TearDown() {
+	if suite.T().Failed() {
+		installer.GatherNFSServerDebuggingInfo(suite.k8sHelper, suite.systemNamespace)
+		installer.GatherNFSServerDebuggingInfo(suite.k8sHelper, suite.namespace)
+		suite.installHelper.GatherAllNFSServerLogs(suite.systemNamespace, suite.namespace, suite.T().Name())
+	}
+	suite.installHelper.UninstallNFSServer(suite.systemNamespace, suite.namespace)
+}
+
+func (suite *NfsSuite) TestNfsServerInstallation() {
+	logger.Infof("Verifying that nfs server pod %s is running", suite.namespace)
+
+	// verify nfs server operator is running OK
+	assert.True(suite.T(), suite.k8sHelper.CheckPodCountAndState("rook-nfs-operator", suite.systemNamespace, 1, "Running"),
+		"1 rook-nfs-operator must be in Running state")
+
+	// verify nfs server instances are running OK
+	assert.True(suite.T(), suite.k8sHelper.CheckPodCountAndState("rook-nfs", suite.namespace, suite.instanceCount, "Running"),
+		fmt.Sprintf("%d rook-nfs pods must be in Running state", suite.instanceCount))
+
+	// determine the nfs operator pod name
+	podNames, err := suite.k8sHelper.GetPodNamesForApp("rook-nfs-operator", suite.systemNamespace)
+	require.NoError(suite.T(), err)
+	require.Equal(suite.T(), 1, len(podNames))
+
+	// verify nfs server storage
+	nfsPVC, err := suite.k8sHelper.GetPVCStatus("default", "nfs-pv-claim")
+	require.NoError(suite.T(), err)
+	assert.Contains(suite.T(), nfsPVC, "Bound")
+
+	defer suite.rwClient.Delete()
+	podList, err := suite.rwClient.CreateWriteClient("nfs-pv-claim")
+	require.NoError(suite.T(), err)
+	assert.True(suite.T(), true, suite.checkReadData(podList))
+}
+
+func (suite *NfsSuite) checkReadData(podList []string) bool {
+	inc := 0
+	var result string
+	var err error
+	for inc < utils.RetryLoop {
+		result, err = suite.rwClient.Read(podList[0])
+		logger.Infof("nfs volume read exited, err: %+v. result: %s", err, result)
+		if err == nil {
+			break
+		}
+		logger.Warning("nfs volume read failed, will try again")
+		inc++
+		time.Sleep(utils.RetryInterval * time.Second)
+	}
+	require.NoError(suite.T(), err)
+	if result == podList[0] || result == podList[1] {
```

Comment:
comment about what this is doing? we only read from `podList[0]` in the for loop above, so it's not immediately obvious what this check is doing without further context